extensions/amp-live-list/amp-live-list.md

```diff
@@ -1 +1,355 @@
+
+The actual action handler may be at a descendant and does not have to be at the
+`update` reference point. the `amp-live-list` component then has an internal
+`update` method than can receive the action. See the
+`on="tap:my-live-list.update"` handler.
+
+Example:
+
+```html
+<amp-live-list id="my-live-list" data-poll-interval="15000" data-max-items-per-page="20">
+  <div update class="outer-container">
+    <div class="inner-container">
+      <button class="btn" on="tap:my-live-list.update">Click me!</button>
+    </div>
+  </div>
+  <div items></div>
+</amp-live-list>
+```
+
+### items
+
+The `items` reference point is where new items are inserted, replaced or removed.
+children of the `items` reference point are required to have an `id` and a
+`data-sort-time` attribute.
+
+### pagination
+
+The `pagination` reference point is where any sort of pagination markup is located.
+We recommend having a small subtree underneath this reference point as the
+`amp-live-list` component will be doing an inline replace of the DOM received from
+the server in case the page count had increased. We don't do any special
+diffing and just outright replace the contents.
+
+## Update Behavior and User Experience (Work in Progress)
+
+When updates are discovered by the client from polling the server document, any
+newly discovered `id`'s from children of the `items` reference point will turn
+the `update` reference point visible. A user action is required for these items
+to be inserted into the client's live DOM. We scroll the top of the
+`amp-live-list` component into view when the reader clicks on the update action.
+
+If a `data-update-time` attribute is present and its value is a number higher
+than the original `data-sort-time` on the attribute, the item will be updated in
+place through a `replaceChild` operation. If a `data-tombstone` attribute is
+present the element's subtree will be emptied out and the item is hidden through
+css (see [amp-live-list.css](https://github.com/ampproject/amphtml/blob/master/extensions/amp-live-list/0.1/amp-live-list.css)).
+
+If a replace or tombstone operation is found but no insert (no new items) is
+found on a poll request, the replace and the tombstone operation will occur
+without the need for a reader's action. The component and the AMP runtime
+will try and maintain the viewport's scroll position but since a replace could
+have components in its subtree that cause a resize (amp-twitter, amp-iframe,
+etc.) after the fact, we discourage having a lot of replace operations through
+the `data-update-time` attribute.
+
+## How Pagination works
+
+As AMP's focus is on performance we recommend the publisher keep a low number of
+items on a single page by limiting the number of children the `items` reference
+point has, as well as setting a good `data-max-items-per-page` value. Neither the client
+nor any cache that may be used, knows anything about the full number of items so
```

Comment:
nor any cache that may be used knows anything about the full number of items, so
(strike comma in middle, add one near end)